gematria/granite/graph_builder_model_inference.cc

```diff
@@ -83,21 +83,34 @@ constexpr std::string_view kInstructionNodeMaskTensorName =
 // `GetInputTensorName` and `GetInputTensorFromName` to create an efficient
 // mapping between tensor and their names at compile time.
 constexpr std::pair<InputTensor, std::string_view> kPossibleInputTensors[]{
-    {InputTensor::kDeltaBlockIndexTensor, kDeltaBlockIndexTensorName},
-    {InputTensor::kGraphNodesTensor, kGraphNodesTensorName},
-    {InputTensor::kGraphEdgesTensor, kGraphEdgesTensorName},
-    {InputTensor::kGraphGlobalsTensor, kGraphGlobalsTensorName},
-    {InputTensor::kGraphReceiversTensor, kGraphReceiversTensorName},
-    {InputTensor::kGraphSendersTensor, kGraphSendersTensorName},
-    {InputTensor::kGraphNEdgeTensor, kGraphNEdgeTensorName},
-    {InputTensor::kGraphNNodeTensor, kGraphNNodeTensorName},
-    {InputTensor::kInstructionNodeMaskTensor, kInstructionNodeMaskTensorName},
+    {InputTensor::kDeltaBlockIndex, kDeltaBlockIndexTensorName},
+    {InputTensor::kGraphNodes, kGraphNodesTensorName},
+    {InputTensor::kGraphEdges, kGraphEdgesTensorName},
+    {InputTensor::kGraphGlobals, kGraphGlobalsTensorName},
+    {InputTensor::kGraphReceivers, kGraphReceiversTensorName},
+    {InputTensor::kGraphSenders, kGraphSendersTensorName},
+    {InputTensor::kGraphNEdge, kGraphNEdgeTensorName},
+    {InputTensor::kGraphNNode, kGraphNNodeTensorName},
+    {InputTensor::kInstructionNodeMask, kInstructionNodeMaskTensorName},
 };
 
 // Graph builder models can have up to these many different input tensors
 // depending on their configuration, as enumerated by `InputTensor`.
 constexpr int kNumPossibleInputTensors = std::size(kPossibleInputTensors);
 
+// Basic checks to ensure the look-up table is consistent with `InputTensor`.
+// The second assertion must be manually updated with the first `InputTensor`
+// and the third assertion with the last `InputTensor` when a new one is added.
+static_assert(static_cast<int>(kPossibleInputTensors[0].first) == 0,
+              "The start of `kPossibleInputTensors` is not aligned with the "
+              "start of `InputTensor`");
+static_assert(static_cast<int>(InputTensor::kDeltaBlockIndex) == 0,
+              "`InputTensor` does not begin enumerating from 0");
+static_assert(
+    static_cast<int>(InputTensor::kInstructionNodeMask) ==
+        kNumPossibleInputTensors - 1,
+    "`kPossibleInputTensors` does not have an entry for every `InputTensor`");
```

Comment:
We could also do something like
```cc
template<typename Container>
constexpr bool IsIncreasing(const Container& entries) {
  const int num_items = std::size(entries);
  for (int i = 1; i < num_items; ++i) {
    if (entries[i - 1].first < entries[i].first) return false;
  }
  return true;
}
static_assert(IsIncreasing(kPossibleInputTensors));
```
to check that the indices are contiguous (there are no duplicates and they are sorted). Together with the other checks, this should cover most of the things that could go wrong.